niche_vlaanderen/flooding.py

```diff
@@ -112,50 +137,60 @@ def _calculate(self, depth, frequency, duration, period):
             subtable_veg = subtable_veg.reset_index()
             # by default we give code 4 (no information/flooding)
             # https://github.com/inbo/niche_vlaanderen/issues/87
-            self._veg[veg_code] = np.full(depth.shape, 4, dtype="int16")
-            self._veg[veg_code][nodata] = -99
+
+            vegi = np.full(depth.shape, 4, dtype=self.dtype)
+            vegi[nodata] = self.nodata
+
             groupby_cols = ["period", "frequency", "duration"]
             for index, subtable in subtable_veg.groupby(groupby_cols):
                 if (period, frequency, duration) == index:
                     subtable.reset_index()
                     for row in subtable.itertuples():
-                        veg = self._veg[veg_code]
+                        veg = vegi
                         veg[row.depth == depth] = row.potential
-                        self._veg[veg_code] = np.maximum(veg, self._veg[veg_code])
-            self._veg[veg_code] = self._veg[veg_code].reshape(orig_shape)
+                        vegi = np.maximum(veg, vegi)
+
+            self._veg[veg_code] = vegi.reshape(orig_shape).astype(self.dtype)
+
+    def read_depth_to_grid(self, file_name):
+        """Read depth file using rasterio to numpy array and set extent
 
-    def calculate(self, depth_file, frequency, period, duration):
+        Parameters
+        ----------
+        file_name : string | Pathlib.Path
+            Path to the file to be read
+        """
+        with rasterio.open(file_name, "r") as dst:
+            band = dst.read(1, masked=True)
+            self._context = SpatialContext(dst)
+
+        # Convert to depth data type and no-data value
+        band = band.astype(_allowed_input["depth"])
+        band = np.ma.masked_array(band.filled(255), mask=band.mask,
```

Comment:
Did you intentionally keep a masked array here?

Reply:
no, thanks for noticing.